AnnService/inc/Core/Common/Dataset.h

```diff
@@ -47,8 +49,10 @@ namespace SPTAG
                     if (data_ != nullptr) memcpy(data, data_, ((size_t)rows) * cols * sizeof(T));
                     else std::memset(data, -1, ((size_t)rows) * cols * sizeof(T));
                 }
-                rowsInBlock = rowsInBlock_;
-                incBlocks.reserve((static_cast<std::int64_t>(capacity_) + rowsInBlock - 1) / rowsInBlock);
+                maxRows = capacity_;
+                rowsInBlockEx = static_cast<SizeType>(ceil(log2(rowsInBlock_)));
+                rowsInBlock = (1 << rowsInBlockEx) - 1;
+                incBlocks.reserve((static_cast<std::int64_t>(capacity_) + rowsInBlock) >> rowsInBlockEx);
```

Comment:
you may need check some edge case like rowInBlock_ == 1, then looks rowsInBlockEx = 0, then rowsInBlock = 0 also, which looks not right

Reply:
ok, you are right, let me fix it